### PR TITLE
fixed percent match weighting

### PR DESCRIPTION
**Pull request by ymatsuda**

### CODE REVIEW

File: shoebox/app/com/keepit/search/query/BooleanQueryWithPercentMatch.scala [hunk 1]
```diff
@@ -76,28 +75,50 @@ class BooleanQueryWithPercentMatch(val disableCoord: Boolean = false) extends LB
...
+
+      override def sumOfSquaredWeights(): Float = {
+        var sum = 0.0d
+        clauses.zip(weights).foreach{ case (c, w) =>
+          val value = w.sumOfSquaredWeights().toDouble
+          if (c.isRequired()) {
+            totalValueOnRequired += sqrt(value).toFloat
+            // if a required clause does not have a scorer, no hit
+            requiredWeights += w
+            sum + value
```
Comment (2is10) [inline shoebox/app/com/keepit/search/query/BooleanQueryWithPercentMatch.scala:92]: `+=` (and on line 98) ?

File: shoebox/app/com/keepit/search/query/BooleanQueryWithPercentMatch.scala [hunk 2]
```diff
@@ -76,28 +75,50 @@ class BooleanQueryWithPercentMatch(val disableCoord: Boolean = false) extends LB
...
+          } else if (c.isProhibited()) {
+            prohibitedWeights += w
+          } else {
+            totalValueOnOptional += sqrt(value).toFloat
+            optionalWeights += ((w, sqrt(value).toFloat))
+            sum + value
+          }
+        }
+        sum.toFloat * getBoost() * getBoost()
```
Comment (2is10) [inline shoebox/app/com/keepit/search/query/BooleanQueryWithPercentMatch.scala:101]: I'm not sure how performance-critical this code is, but when I see repeated calls, I can't help but suggest stashing the result in a local val (for both `sqrt(value).toFloat` and `getBoost()`). Is the scala compiler smart enough to deduce that both invocations will return the same value and have no side effects, and so optimize the second one away? I'd be really impressed.

Comment (ymatsuda) [inline shoebox/app/com/keepit/search/query/BooleanQueryWithPercentMatch.scala:101]: I doubt the compiler is that smart. :)
I will fix it.

Comment (ymatsuda) [inline shoebox/app/com/keepit/search/query/BooleanQueryWithPercentMatch.scala:101]: I mean my code, not the compiler.

**Pull request merged**